core/sdk/src/framework/ripple_contract.rs

```diff
@@ -121,7 +133,41 @@ impl RippleContract {
     /// This method gets the clear string of the contract without the quotes added
     /// by serde deserializer.
     pub fn as_clear_string(self) -> String {
-        SerdeClearString::as_clear_string(&self)
+        let contract = SerdeClearString::as_clear_string(&self);
+        if let Some(adjective) = self.get_adjective() {
+            format!("{}.{}", adjective, contract)
+        } else {
+            contract
+        }
+    }
+
+    pub fn get_adjective(&self) -> Option<String> {
+        if let Self::Storage(s) = self {
+            return Some(s.as_string());
+        }
+        if let Self::Session(s) = self {
+            return Some(s.as_string());
+        }
+
+        None
+    }
+
+    pub fn from_adjective_string(contract: &str, adjective: &str) -> Option<Self> {
+        let adjective = format!("\"{}\"", adjective);
+        match contract {
+            "storage" => {
+                if let Ok(v) = serde_json::from_str::<StorageAdjective>(&adjective) {
```

Comment:
cxIf this does not deserialise and we get an Err should we at least log it? At that point we will return None and we may never know there is a problem.